crates/chia-sdk-driver/src/primitives/mips/memo.rs

```diff
@@ -0,0 +1,572 @@
+mod parsed_member;
+mod parsed_restriction;
+mod parsed_wrapper;
+
+use chia_consensus::opcodes::{
+    CREATE_COIN_ANNOUNCEMENT, CREATE_PUZZLE_ANNOUNCEMENT, RECEIVE_MESSAGE, SEND_MESSAGE,
+};
+pub use parsed_member::*;
+pub use parsed_restriction::*;
+pub use parsed_wrapper::*;
+
+use chia_bls::PublicKey;
+use chia_protocol::Bytes32;
+use chia_sdk_types::{
+    puzzles::{
+        BlsMember, BlsTaprootMember, EnforceDelegatedPuzzleWrappers, FixedPuzzleMember,
+        Force1of2RestrictedVariable, K1Member, K1MemberPuzzleAssert, PasskeyMember,
+        PasskeyMemberPuzzleAssert, PreventConditionOpcode, R1Member, R1MemberPuzzleAssert,
+        SingletonMember, Timelock, FORCE_ASSERT_COIN_ANNOUNCEMENT_PUZZLE_HASH,
+        FORCE_COIN_MESSAGE_PUZZLE_HASH, PREVENT_MULTIPLE_CREATE_COINS_PUZZLE_HASH,
+    },
+    Mod,
+};
+use chia_secp::{K1PublicKey, R1PublicKey};
+use clvm_traits::{apply_constants, FromClvm, ToClvm};
+use clvm_utils::TreeHash;
+use clvmr::{Allocator, NodePtr};
+
+use crate::DriverError;
+
+#[derive(ToClvm, FromClvm)]
+#[apply_constants]
```

Comment:
reading the `apply_constants` macro, it's really quite hard to tell what it does. It's kind of unrelated to this PR, but this clvm_traits macro should be documented

Reply:
It removes fields declared with `#[clvm(constant)]` from the actual struct, since they're only used by the CLVM encoding macros. Therefore, this has to be applied after `ToClvm, FromClvm` and before any other macros which derive the final struct.